src/hotspot/share/opto/vectorization.hpp

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2023, Arm Limited. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#ifndef SHARE_OPTO_VECTORIZATION_HPP
+#define SHARE_OPTO_VECTORIZATION_HPP
+
+#include "opto/node.hpp"
+#include "opto/loopnode.hpp"
+
+// ----------------------------------VPointer----------------------------------
+// Information about an address for dependence checking and vector alignment
```

Comment:
I am fine with using VPointer name but I think you need to add more description here to avoid confusion.

May be add a general description what is in this file: "for shared logics and utilities for C2's auto-vectorization".

Reply:
Done